src/mongo/util/processinfo.h

```diff
@@ -151,10 +151,9 @@ class ProcessInfo {
     /**
      * Get extra system stats
      */
-    void appendSystemDetails(BSONObjBuilder& details) const {
-        details.append(StringData("extra"), sysInfo()._extraStats.copy());
+    static const BSONObj getSystemDetails() {
+        return sysInfo()._extraStats.copy();
     }
```

Comment:
```suggestion
    void appendSystemDetails(BSONObjBuilder& details) const {
        details.appendElements(_sysInfo()._extraStats);
    }

```

Reply:
I'm swiss on doing it this way or the other.  Either way there's a copy of the object contents being burned (though agree on `const` being wrong here).

Reply:
I agree that we're wasting one and object copy either way.

But I still like my way better, because

- we're only changing the internals of the function, and

- The object is wasted only because the caller was being lazy and chose to waste it. But a smarter caller could avoid making a copy. They have a choice. In the original pull request, there was no choice the copy was required by the calling convention.

In some sense, this is medieval monastic debate, but we don't need to change API so i lean towards not doing so

Reply:
Will use
`details.appendElements(sysInfo()._extraStats);`
vs:
`details.appendElements(_sysInfo()._extraStats);`